app/models/concerns/system_like.rb

```diff
@@ -19,4 +19,24 @@ def compliant
     end
     result
   end
+
+  def total_rules(profile = nil)
+    return profile.results(self) if profile.present?
+    profiles.map do |p|
+      p.results(self)
+    end.flatten
+  end
+
+  def rules_passed(profile = nil)
+    total_rules(profile).count { |result| result }
+  end
+
+  def rules_failed(profile = nil)
+    total_rules(profile).count(&:!)
+  end
+
+  def compliance_score
+    score = (100 * (rules_passed.to_f / (rules_passed + rules_failed)))
+    score.nan? ? 0.0 : score
```

Comment:
If a host has no rule results, it doesn't make sense to say 0 or 100. I wonder if we should just return nil?

Reply:
I agree, normally that situation should only happen if there was some kind of db corruption or bug.